### PR TITLE
Cleaning up API docs

### LINKED ISSUE #42: docs are missing content from MDAnalysis.analysis.encore
## Expected behavior ##

The MDAKit docs contain all information that is available in MDAnalysis.analysis.encore.

There's a section before "API Documentation" that tells me about the general purpose and underlying ideas.

## Actual behavior ##

The top level introduction to ENCORE from https://docs.mdanalysis.org/stable/documentation_pages/analysis/encore.html is missing. There's no overview that explains what ENCORE does.

## Suggest changes
- transfer content from the orginal docs to new encore docs
- arrange in narrative order (similar to presentation in https://docs.mdanalysis.org/stable/documentation_pages/analysis/encore.html )

## Current environment ##

- Which version are you using? 0.1.0

### LINKED ISSUE #43: [DOCS] links to functions/classes are non-existent
## Expected behavior ##

When digging through the API docs I should be able to get to a full documentation of classes and functions.

## Actual behavior ##

Clicking any class name in a summary table such as `bootstrapped_matrix` in https://www.mdanalysis.org/mdaencore/autosummary/mdaencore.bootstrap.html#module-mdaencore.bootstrap , which is styled as a link, fails to go anywhere: there's no underlying link.

## Code to reproduce the behavior ##

E.g. https://www.mdanalysis.org/mdaencore/autosummary/mdaencore.bootstrap.html#module-mdaencore.bootstrap and click any of the orange links in the table:

image

However, this is true for most links.

For a few classes, such as [mdaencore.clustering.ClusterCollection](https://www.mdanalysis.org/mdaencore/autosummary/mdaencore.clustering.ClusterCollection.html#mdaencore.clustering.ClusterCollection) the links resolve but then the links for the **Methods** are not present.

## Current environment ##

- Which version are you using? 0.1.0 docs on RTD

### PR DESCRIPTION
**Pull request by ianmkenney**
Changes made in this Pull Request:

 - API docs recreated to look more like original documentation from the core library

Fixes #42 
Fixes #43 

PR Checklist
------------
 - [ ] Tests?
 - [x] Docs?
 - [ ] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by orbeckst** [comment]
Note: add link to https://userguide.mdanalysis.org/stable/examples/analysis/trajectory_similarity/clustering_ensemble_similarity.html

**Review by orbeckst — CHANGES_REQUESTED** [review]
Looking really good, few things

1. move intro from API to landing page, put kit history into last paragraph
2. link to UG Note: add link to https://userguide.mdanalysis.org/stable/examples/analysis/trajectory_similarity/clustering_ensemble_similarity.html on landing page (might eventually be merged but not right now)
3. add quick intro paragraph to API section

**Comment by ianmkenney** [comment]
I referenced the user guide, but it's not quite 1:1 since the user guide references the core library implementation. I suppose I'll just a note that since the `encore` module hasn't been removed, that you need to modify the imports. Want to just state the explicitly here.

File: docs/source/api.rst [hunk 1]
```diff
@@ -1,10 +1,28 @@
+=================
 API Documentation
 =================
 
-This documentation details the modules, classes, and functions in ``mdaencore``.
+This section contains an overview of the ``mdaencore`` API.
+For higher level examples of how to use the software, see the `ENCORE section <https://userguide.mdanalysis.org/stable/examples/analysis/trajectory_similarity/clustering_ensemble_similarity.html>`_ of the MDAnalysis User Guide.
+Note that the user guide imports the ENCORE implementation in the core library, as the current version of MDAnalysis still has this functionality.
+To use these examples, import ``mdaencore`` rather than ``MDAnalysis.analysis.encore``.
```
Comment (orbeckst) [inline docs/source/api.rst:8]: Make these 3 lines a `.. see also::` block on the landing page. The API docs can be terser, the landing page should have all the "how do I" / "what does this look like in practice" information.

**Review by orbeckst — CHANGES_REQUESTED** [review]
Add a page for the references. Clicking a reference does not go anywhere for me (eg on the first page "The package contains implementations of similarity measures between protein ensembles described in [[1]](https://mdaencore--49.org.readthedocs.build/en/49/#footcite-lindorfflarsen2009).") Something is not working with footcite.

Also move the (good!) paragraph with the link to the UG to the landing page.

File: docs/source/api.rst (hunk 1 above)
Comment (ianmkenney) [reply docs/source/api.rst]: Moved!

**Comment by orbeckst** [comment]
I don't see the changes in the RTD build (neither see also box nor working refs). Did you check your build?

image

**Comment by ianmkenney** [comment]
> I don't see the changes in the RTD build (neither see also box nor working refs). Did you check your build?
> 
> https://mdaencore--49.org.readthedocs.build/en/49/#footcite-lindorfflarsen2009
> 
> image

@orbeckst Do a hard refresh

**Review by orbeckst — APPROVED** [review]

**Pull request merged**